spec/index.bs

```diff
@@ -2912,70 +2912,100 @@ The {{GPUMapMode}} flags determine how a {{GPUBuffer}} is mapped when calling
+                        - If |mode| contains {{GPUMapMode/WRITE}} then |this|.{{GPUBuffer/usage}} must contain {{GPUBufferUsage/MAP_WRITE}}.
+
+                        Issue: Do we validate that |mode| contains only valid flags?
+                    </div>
 
-                1. Record a validation error on the current scope.
-                1. Return [=a promise rejected with=] an {{OperationError}} on the [=Device timeline=].
+                    Then:
 
-            1. Let |p| be a new {{Promise}}.
-            1. Set |this|.{{GPUBuffer/[[mapping]]}} to |p|.
-            1. Set |this|.{{GPUBuffer/[[state]]}} to [=buffer state/mapping pending=].
-            1. Set |this|.{{GPUBuffer/[[map_mode]]}} to |mode|.
-            1. Enqueue an operation on the default queue's [=Queue timeline=] that will execute the following:
+                    1. Issue the following steps on |contentTimeline|:
 
-                <div class=queue-timeline>
-                    1. If |this|.{{GPUBuffer/[[state]]}} is [=buffer state/mapping pending=]:
+                        <div data-timeline=content>
+                            [=Content timeline=] steps:
 
-                        1. Let |m| be a new {{ArrayBuffer}} of size |rangeSize|.
+                            1. [=Reject=] |p| with an {{OperationError}}.
+                        </div>
 
-                            Note: This is just a memory allocation; this {{ArrayBuffer}} object is not exposed directly.
-                        1. Set the content of |m| to the content of |this|'s allocation starting at offset |offset| and for |rangeSize| bytes.
-                        1. Set |this|.{{GPUBuffer/[[mapping]]}} to |m|.
-                        1. Set |this|.{{GPUBuffer/[[state]]}} to [=buffer state/mapped=].
-                        1. Set |this|.{{GPUBuffer/[[mapping_range]]}} to <code>[|offset|, |offset| + |rangeSize|]</code>.
-                        1. Set |this|.{{GPUBuffer/[[mapped_ranges]]}} to `[]`.
+                    1. [$Generate a validation error$].
+                    1. Return.
 
-                    1. Resolve |p|.
-                </div>
-            1. Return |p|.
+                1. Once all <span data-timeline=queue>currently-enqueued operations using |this| (on any queue)</span>
+                    are completed, issue the subsequent steps on |contentTimeline|.
+
+                    Note: Since the buffer is mapped, its contents cannot change between now and {{GPUBuffer/unmap()}}.
+
+                    <!-- When we have multi-queue, this may be better described using queue-transfer language. -->
+            </div>
+            <div data-timeline=device>
+                [=Device timeline=] steps:
+
+                1. Let |dataForMappedRegion| be the contents of |this| starting at offset |offset|, for |rangeSize| bytes.
+                1. Issue: Set any state here as needed (if any).
+                1. Issue the subsequent steps on |contentTimeline|:
```

Comment:
Can this `|contentTimeline| be wrapped in a `<span data-timeline=content>`? I really like how the other timeline transitions underline the timeline name/variable with the appropriate color. (Same thing applies for the content transition right after validation)